src/main/java/subway/LineService.java

```diff
@@ -24,27 +28,52 @@ public LineResponse createLine(LineRequest request) {
         Station upStation = stationRepository.findById(request.getUpStationId()).orElseThrow(() -> new NoSuchFieldError("해당 지하철역이 없습니다."));
         Station downStation = stationRepository.findById(request.getDownStationId()).orElseThrow(() -> new NoSuchFieldError("해당 지하철역이 없습니다."));
         lineStationRepository.save(new LineStation(0, line, upStation));
-        lineStationRepository.save(new LineStation(request.getDistance() - 1, line, downStation));
+        lineStationRepository.save(new LineStation(1, line, downStation));
         return createLineResponse(line, upStation, downStation);
     }
 
-    private LineResponse createLineResponse(Line line, Station upStation , Station downStation) {
-        return new LineResponse(line.getId(),
-                line.getName(),
-                line.getColor(),
-                List.of(new StationResponse(upStation.getId(), upStation.getName()), new StationResponse(downStation.getId(), downStation.getName())));
-    }
 
+    @Transactional(readOnly = true)
     public List<LineResponse> findLines() {
         List<Line> lines = lineRepository.findAll();
         return lines.stream().map(this::createLineResponse).collect(Collectors.toList());
     }
 
+    @Transactional(readOnly = true)
     public LineResponse findLine(Long id) {
         Line line = lineRepository.findById(id).orElseThrow(() -> new RuntimeException("해당 아이디의 지하철 노선이 없습니다."));
         return createLineResponse(line);
     }
 
+    @Transactional
+    public void updateLine(Long id, LineRequest request) {
+        Line line = lineRepository.findById(id).orElseThrow(() -> new RuntimeException("해당 아이디의 지하철 노선이 없습니다."));
+        line.updateLine(request.getName(), request.getColor());
+    }
+
+    @Transactional
+    public void deleteLine(Long id) {
+        lineRepository.deleteById(id);
+    }
+
+    @Transactional
+    public LineResponse addLine(Long id, AddLineRequest dto) {
+        // 새로운 구간의 상행역은 해당 노선에 등록되어있는 하행 종점역이여야한다.
+        Station upStation = stationRepository.findById(dto.getUpStationId()).orElseThrow(() -> new RuntimeException("상행선역 지하철을 먼저 등록해주세요"));
+        Station downStation = stationRepository.findById(dto.getDownStationId()).orElseThrow(() -> new RuntimeException("하행선역 지하철을 먼저 등록해주세요"));
+        Line line = lineRepository.findById(id).orElseThrow(() -> new RuntimeException("기존에 등록되어 있는 지하철 노선이 아닙니다. 다시 확인해주세요."));
+        line.isAddableLine(upStation, downStation);
+        lineStationRepository.save(new LineStation(line.getLineStations().size(), line, downStation));
```

Comment:
```suggestion
        line.addLineStation(upStation, downStation, dto.getDistance());
```
구간을 생성해주는 로직을 line으로 모두 옮겨 lineStation을 생성해주는 행위도 LineService가 아닌 도메인에서 할 수 있도록 해도 좋을 것 같습니다 :) 
lineStation을 저장하는 것은 LineService에서 굳이 하지 않아도 더티 체킹으로 인해 자연스럽게 될 수 있습니다.

Reply:
아.. 이런점을 생각하지 못했네요 ㅠ 업무에서도 보니 지금까지 도메인 중심이 아니라 서비스 안에서 레포지토리 호출을 통해 지금까지 처리한 것을 깨달았습니다 감사합니다.